### ISSUE TITLE
Previous release no longer available on debian?

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/bazel#4947]
I'm trying to make a minor change to https://github.com/alexeagle/ngcontainer
which installs a pinned version of Bazel. Building this docker image now fails with:

E: Version '0.9.0' for 'bazel' was not found
The command '/bin/sh -c BAZEL_VERSION="0.9.0"  && wget -q -O - https://bazel.build/bazel-release.pub.gpg | apt-key add -  && echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" > /etc/apt/sources.list.d/bazel.list  && apt-get update  && apt-get install -y bazel=$BAZEL_VERSION  && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100

Is it intended that older releases are no longer installable? This makes it hard to make reproducible builds.

**Comment by philwo** [comment]
Unfortunately this is currently the case - but I wouldn't call it "intended". ;)

The files are all still there on GCS, however the Debian repo metadata doesn't know about them anymore, as it is regenerated locally once per release and only for the latest deb versions.

I think we can fix this. I'll look at it after my vacation.

**Comment by kamahen** [comment]
I upgraded bazel to 0.25.0, which broke things, but couldn't easily downgrade because the previous version isn't in the PPA:
```
$ apt list -a bazel
Listing... Done
bazel/unknown,now 0.25.0 amd64 [installed]
```

Why do you remove the old versions?

**Comment by philwo** [comment]
I'm working on fixing this right now. We will provide all older Bazel versions in our apt-get repository in addition to the latest one and probably also make it possible to install versions in parallel (e.g. Bazel 0.29.1 can be installed side-by-side as `/usr/bin/bazel-0.29.1`).

Comments on the latter idea are welcome, whether that would be useful for you.

**Comment by philwo** [comment]
@meteorcloudy fixed this in https://github.com/bazelbuild/bazel/commit/31627057454717d2216da0791969207569a1f423. :) Our apt repository now contains all stable version released since 1.0.0 and will continue to do so in the future
